azure/services/virtualnetworks/virtualnetworks.go

```diff
@@ -150,19 +154,29 @@ func (s *Service) IsManaged(ctx context.Context) (bool, error) {
 	ctx, _, done := tele.StartSpanWithLogger(ctx, "virtualnetworks.Service.IsManaged")
 	defer done()
 
+	// Report errors if the client is throttled.
+	if s.RetryAfter.Reader.After(time.Now()) {
```

Comment:
should this also be removed?

Reply:
Note: we have a better way to do this with controllers by scheduling a the controller to requeue on the retryAfter time, but let's do that in #2572 and keep this simple for now